specifyweb/frontend/js_src/lib/components/QueryBuilder/FieldFilter.tsx

```diff
@@ -243,34 +479,111 @@ function SingleField({
 
   readonly enforceLengthLimit: boolean;
 }): JSX.Element {
+  if (parser.type === 'date') {
+    return (
+      <DateQueryInputField
+        currentValue={currentValue}
+        fieldName={fieldName}
+        label={label}
+        parser={parser}
+        onChange={handleChange}
+      />
+    );
+  }
+  if (
+    terminatingField?.isRelationship === false &&
+    terminatingField.name === 'name' &&
+    terminatingField.model.name === 'SpecifyUser'
+  ) {
+    return (
+      <SpecifyUserAutoComplete
+        startValue={currentValue}
+        onChange={handleChange}
+      />
+    );
+  }
   return (
     <QueryInputField
-      currentValue={filter.startValue}
+      currentValue={currentValue}
       fieldName={fieldName}
       label={label}
+      listInput={listInput}
       parser={parser}
       pickListItems={pickListItems}
       onChange={handleChange}
     />
   );
 }
 
+function SpecifyUserAutoComplete({
+  startValue,
+  onChange: handleChange,
+}: {
+  readonly startValue: string;
+  readonly onChange: ((newValue: string) => void) | undefined;
+}): JSX.Element {
+  const valueRef = React.useRef<string>(startValue);
+  const items = [
+    {
+      label: 'Current User',
```

Comment:
This must be localized!
Try not putting unlocalized strings in the code even in development or otherwise you would forget to change them